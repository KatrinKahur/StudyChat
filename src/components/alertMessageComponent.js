import {Pressable, Text} from "react-native";
import {Feather} from "@expo/vector-icons";

export const AlertMessageComponent = ({message, chosenStyle = "info", setMessageCallback}) => {
    if(message === null){
        return (<></>)
    }

    if(chosenStyle !== "danger" && chosenStyle !== "success")
        chosenStyle = "info";

    return (
        <Pressable
            testID="message-container"
            onPress={() => setMessageCallback(null)}
            style={{...styles.box.base, ...styles.box[chosenStyle]}}>
            <Text style={{...styles.text.base, ...styles.text[chosenStyle]}}>{message}
                <Feather name="x" size={styles.text.base.fontSize} color={styles.text[chosenStyle].color} /></Text>
        </Pressable>
    )
}

const styles = {
    text: {
        info: {
            color: "#31708f"
        },
        danger: {
            color: "#a94442"
        },
        success: {
            color: "#3c763d"
        },
        base: {
            fontSize: 20
        }
    },
    box: {
        info: {
            backgroundColor: "#d9edf7",
            borderColor: "#bce8f1"
        },
        danger: {
            backgroundColor: "#f2dede",
            borderColor: "#ebccd1"
        },
        success: {
            backgroundColor: "#dff0d8",
            borderColor: "#d6e9c6"
        },
        base: {
            zIndex: 9999,
            position: "absolute",
            top: 10,
            borderWidth: 1,
            paddingVertical: 20,
            paddingLeft: 20,
            paddingRight: 10,
            borderRadius: 5
        }
    }
}