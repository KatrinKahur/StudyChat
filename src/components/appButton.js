import {Pressable, Text, Platform} from "react-native";
import React from "react";

export const AppButton = ({title, onPress, testID=null}) => {
    return(
        <Pressable
            testID={testID} style={styles.button}
            onPress={onPress}>
            <Text style={styles.buttonTitle}>{title}</Text>
        </Pressable>
    )
}

const styles = {
    button: {
        width: Platform.OS === 'web'? "15%" : "55%",
        minWidth: Platform.OS === 'web'? 340 : 0,
        backgroundColor: `#1e90ff`,
        paddingVertical: 17,
        marginTop: 10,
        marginBottom: 15,
        alignItems: "center",
        borderRadius: 5
    },
    buttonTitle: {
        fontSize:  Platform.OS === 'web'? 20 : 15,
        color: `#ffffff`
    }
}