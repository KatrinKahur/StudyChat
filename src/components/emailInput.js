import { Platform, StyleSheet, TextInput, View } from "react-native";
import { FontAwesome } from "@expo/vector-icons";

export const EmailInput = ({ onChangeText, onSubmitEditing, value }) => {
  return (
    <View style={styles.container}>
      <FontAwesome name="user" size={24} color="black" />
      <TextInput
          testID="emailInput"
          style={styles.input}
          value={value}
          onChangeText={onChangeText}
          onSubmitEditing={onSubmitEditing}
          placeholder="Email.."
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    marginTop: Platform.OS === "web" ? 10 : 5,
    paddingRight: 10,
    paddingTop: 20,
    paddingBottom: 5,
    borderBottomWidth: 1,
    marginVertical: 10,
  },
  input: {
    width: "100%",
    fontSize: Platform.OS === "web" ? 20 : 17,
    paddingLeft: 15,
    paddingVertical: 3,
    paddingRight: 15,
    overflow: "hidden",
  },
});
