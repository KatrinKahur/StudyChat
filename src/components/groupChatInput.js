import {TextInput, View, StyleSheet, Platform} from "react-native";
import {FontAwesome} from "@expo/vector-icons";
import React from "react";

export const GroupChatInput = ({ onChangeText, value }) => {
    return (
      <View style={styles.container}>
        <FontAwesome name="long-arrow-right" size={24} color="black" />
        <TextInput
          testID="groupChatInput"
          style={styles.input}
          value={value}
          onChangeText={onChangeText}
          placeholder="Name of chat"
        />
      </View>
    );
  };
  
  const styles = StyleSheet.create({
    container: {
      flexDirection: "row",
      marginTop: Platform.OS === "web" ? 10 : 5,
      paddingRight: 10,
      paddingTop: 20,
      paddingBottom: 5,
      borderBottomWidth: 1,
      marginVertical: 10,
    },
    input: {
      width: "100%",
      fontSize: Platform.OS === "web" ? 20 : 17,
      paddingLeft: 15,
      paddingVertical: 3,
      paddingRight: 15,
      overflow: "hidden",
    },
  });