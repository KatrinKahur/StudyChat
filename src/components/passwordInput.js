import {TextInput, View, StyleSheet, Platform} from "react-native";
import {FontAwesome5} from "@expo/vector-icons";
import React from "react";

export const PasswordInput = ({onChangeText, onSubmitEditing, value}) => {
    return(
        <View style={styles.container}>
            <FontAwesome5 name="lock" size={24} color="black" />
            <TextInput
                testID="passwordInput"
                style={styles.input} value={value}
                secureTextEntry={true}
                onSubmitEditing={onSubmitEditing}
                onChangeText = {onChangeText} placeholder="Password.."/>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        marginTop: Platform.OS === 'web'? 10 : 5,
        paddingRight: 10,
        paddingTop: 20,
        paddingBottom: 5,
        borderBottomWidth: 1,
        marginVertical: 10
    },
    input: {
        width: "100%",
        fontSize: Platform.OS === 'web'? 20 : 17,
        paddingLeft: 15,
        paddingVertical: 3,
        paddingRight: 15,
        overflow: "hidden"
    }
})