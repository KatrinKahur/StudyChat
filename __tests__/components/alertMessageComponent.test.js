import {AlertMessageComponent} from "../../src/components/alertMessageComponent";
import renderer from 'react-test-renderer';
import {render, cleanup, fireEvent} from "@testing-library/react-native";
import { toHaveStyle, toHaveTextContent } from "@testing-library/jest-native";
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom'

beforeEach(() => {
    cleanup();
})

describe('AlertMessageComponent', () => {
    expect.extend({ toHaveStyle, toHaveTextContent });
    it('should render correctly', () => {
        let testMessage = "Random message";
        const setTestMessage = (message) => {
            testMessage = message;
        }
        let style = "danger"
        const testProps = {
            message: testMessage,
            chosenStyle: style,
            setMessageCallback: setTestMessage
        }

        const { toJSON } = renderer.create(
            <AlertMessageComponent {...testProps} />
        )
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render "info"-style if chosenStyle not specified', () => {
        let testMessage = "Random message";
        const setTestMessage = (message) => {
            testMessage = message;
        }
        let style = ""
        const testProps = {
            message: testMessage,
            chosenStyle: style,
            setMessageCallback: setTestMessage
        }
        const { getByText, getByTestId } = render(<AlertMessageComponent
            {...testProps} />);
        expect(getByText(testMessage)).toHaveStyle({
            color: "#31708f"
        })
        expect(getByTestId('message-container')).toHaveStyle({
            backgroundColor: "#d9edf7",
            borderColor: "#bce8f1"
        });
    });

    it('should render an empty view if message property is null', () => {
        let testMessage = null;
        const setTestMessage = (message) => {
            testMessage = message;
        }
        let style = ""
        const testProps = {
            message: testMessage,
            chosenStyle: style,
            setMessageCallback: setTestMessage
        }
        const { queryByTestId } = render(<AlertMessageComponent
            {...testProps} />);
        expect(queryByTestId('message-container')).toBeNull();
    });

    it('should become invisible when pressed on', async () => {
        await act(async () => {
            let testMessage = "Random message";
            const setTestMessage = (message) => {
                testMessage = message;
            }
            let style = ""
            const testProps = {
                message: testMessage,
                chosenStyle: style,
                setMessageCallback: setTestMessage
            }
            const { queryByTestId } = render(<AlertMessageComponent
                {...testProps} />);
            fireEvent.press(queryByTestId('message-container'));
            expect(testMessage).toBeNull();
        })
    });
})