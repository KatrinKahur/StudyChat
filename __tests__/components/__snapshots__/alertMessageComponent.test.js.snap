// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessageComponent should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#f2dede",
      "borderColor": "#ebccd1",
      "borderRadius": 5,
      "borderWidth": 1,
      "paddingLeft": 20,
      "paddingRight": 10,
      "paddingVertical": 20,
      "position": "absolute",
      "top": 10,
      "zIndex": 9999,
    }
  }
  testID="message-container"
>
  <Text
    style={
      Object {
        "color": "#a94442",
        "fontSize": 20,
      }
    }
  >
    Random message
    <Text />
  </Text>
</View>
`;
